tests/test.cpp

```diff
@@ -1667,6 +1667,13 @@ void ErrorTest() {
   TestError("enum X:byte { Y, Y }", "value already");
   TestError("enum X:byte { Y=2, Z=2 }", "unique");
   TestError("table X { Y:int; } table X {", "datatype already");
+  TestError("table X { A:bool; } table Y { } union X {",
+            "enum clashes with datatype");
+  TestError("union X { X, Y } table X {", "datatype clashes with enum");
+  TestError("namespace A; table X { } namespace A; union X {",
```

Comment:
Can we test the inverse condition, where the names are the same but in different namespaces?

Reply:
I agree that would be a good test but I couldn't seem how to assert that a schema throws no error. Is there a place to do that @vglavnyy ?

Reply:
There is the `EnumNamesTest()` test. It might be used for new tests.
However, I think it would be better to add the new independent test case function.